### PR TITLE
DROOLS-4319: [DMN Designer] Search feature - Implement search feature in the GDT graphs editor

### PR DESCRIPTION
**Pull request by danielzhe**
Part of an ensemble. This is are new PRs since the older was broken.
https://github.com/kiegroup/kie-wb-common/pull/2801
https://github.com/kiegroup/appformer/pull/759
https://github.com/kiegroup/drools-wb/pull/1206****

### CODE REVIEW

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/editor/GuidedDecisionTableGraphEditorPresenter.java [hunk 1]
```diff
@@ -204,6 +222,84 @@ public void init() {
...
+
+                    searchableElement.setWidget(widget);
+                    searchableElements.add(searchableElement);
+                }
+            }
+        }
+        return searchableElements;
+    }
+
+    GuidedDecisionTableView getWidgetToModel(GuidedDecisionTable52 model) {
```
Comment (jomarko) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/editor/GuidedDecisionTableGraphEditorPresenter.java]: Could we please use `final GuidedDecisionTable52 model`? Just for consistency.

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/editor/GuidedDecisionTableGraphEditorPresenter.java [hunk 2]
```diff
@@ -204,6 +222,84 @@ public void init() {
...
+
+        final Optional<GridWidget> widget = modeller.getView()
+                .getGridWidgets()
+                .stream()
+                .filter(w -> (w instanceof GuidedDecisionTableView)
+                        && ((GuidedDecisionTableView) w).getPresenter().getModel().equals(model))
+                .findFirst();
+
+        return (GuidedDecisionTableView) widget.get();
```
Comment (jomarko) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/editor/GuidedDecisionTableGraphEditorPresenter.java]: I was able to write as single statement:

```
return modeller.getView()
                .getGridWidgets()
                .stream()
                .filter(w -> w instanceof GuidedDecisionTableView)
                .map(w -> (GuidedDecisionTableView) w)
                .filter(w -> Objects.equals(w.getPresenter().getModel(), model))
                .findFirst()
                .get();
```

What do you think about it?

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/test/java/org/drools/workbench/screens/guided/dtable/client/editor/GuidedDecisionTableGraphEditorPresenterTest.java [hunk 3]
```diff
@@ -1674,7 +1677,7 @@ private GuidedDecisionTableGraphEditorPresenter makePresenter() {
                                                            columnsPage,
                                                            saveAndRenameCommandBuilder,
                                                            alertsButtonMenuItemBuilder,
-                                                           downloadMenuItemBuilder);
+                                                           downloadMenuItemBuilder, editorSearchIndex, searchBarComponent, searchableElementFactory);
```
Comment (jomarko) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/test/java/org/drools/workbench/screens/guided/dtable/client/editor/GuidedDecisionTableGraphEditorPresenterTest.java]: Just for consistency, could we keep one argument per line please?

**Comment by danielzhe** [comment]
@jomarko Changes applied! Thank you!

**Comment by karreiro** [comment]
Jenkins please retest this.

**Comment by karreiro** [comment]
@danielzhe I cannot build this branch. I think the methods `withMinX` and `withMinY` were removed from `GridHighlightHelper`, but this PR is not updated.

**Comment by danielzhe** [comment]
@karreiro Fixed. Thank you.

**Pull request merged**